Lint fixes and enhancements

jbaptperez opened the pull request:
A couple of modifications:

1. Fixes lint issues:
    1. flake8,
    2. pylint (factorizes duplicated code into a `pytests/common.py` file),
2. Fixes CONTRIBUTING.md:
    1. Typos,
    2. Updates URL of signing request documentation (--signoff instead of --gpg-sign),
    3. Formats Markdown.

Lint fixing commits are independents.

As I added the `pytests/common.py` source file to factorise some duplicated code, the `pre-commit run --all-files` command starts require an initial call to `export PYTHONPATH=$PWD` where `$PWD` is the root directory of the repository.

Is really `# pylint: disable=import-error` a correct way to handle this kind of issue?
I did not find a proper and portable way to solve this problem directly in `.pylintrc`, but I'm open to suggestions.

p12tic reviewed (changes requested):
Thanks for the PR.

In the documentation, please keep line length to around 100 chars or less.
 
The unit tests have migrated from pylint to unittest, so these changes are probably no longer relevant.

I understand a lot of time have passed, so you probably won't have time to go back to the PR. Maybe someone else can use the review to help extract useful stuff from the PR.

jbaptperez commented:
@p12tic

> Thanks for the PR.
> 
> In the documentation, please keep line length to around 100 chars or less.

Done.

> The unit tests have migrated from pylint to unittest, so these changes are probably no longer relevant.
> 
> I understand a lot of time have passed, so you probably won't have time to go back to the PR. Maybe someone else can use the review to help extract useful stuff from the PR.

I removed outdated fixes, rebased my branch on `main` and updated the PR.

However, the pytest to unitest and ruff migrations look not to be completed and I think `README.md` and `CONTRIBUTING.md` (or even `setup.py`) have to be updated accordingly.

p12tic commented:
> However, the pytest to unitest and ruff migrations look not to be completed and I think README.md and CONTRIBUTING.md (or even setup.py) have to be updated accordingly.

Thanks, indeed..
